modules/hal_nxp/usb/usb_device_config.h

```diff
@@ -39,7 +39,13 @@
 /* Whether device is self power. 1U supported, 0U not supported */
 #define USB_DEVICE_CONFIG_SELF_POWER (1U)
 
-#define DT_DRV_COMPAT nxp_mcux_usbd
+#if DT_HAS_COMPAT_STATUS_OKAY(nxp_lpcip3511)
```

Comment:
I'm assuming you are absolutely sure it will always only ever be one or the other. If so, feel free to resolve this comment.

If not, can you add a BUILD_ASSERT that the devicetree does not have nodes with both compatibles in it, just to document the assumption?

Something like `BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(nxp_lpcip3511) + DT_NUM_INST_STATUS_OKAY(nxp_ehci), ...)`.